### PR TITLE
Match State Flow

### PR DESCRIPTION
**Pull request by PepperLola**
### Description
Integrates the scoreboard and timer into match mode. There is now a phase to add/edit scoring zones, and upon closing that panel a new confirm one will pop up. Confirming match start puts the game into auto mode, and after 15 seconds (displayed on the scoreboard), there is a 3s transition period in which robot inputs are frozen and then teleop starts. After teleop ends, the match review phase is entered.

### Objectives
- [X] Show an accurate timer based on the match mode state
- [X] Add state to edit scoring zones
- [x] Add transition period that freezes robot inputs

### Testing Done
- Gone through the process of entering match mode, configuring, playing a match, and then getting to the match review section
- Clicked a bunch of buttons while in each stage

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1553)
[JIRA Issue](https://jira.autodesk.com/browse/AARD-1560)

### CODE REVIEW

File: engine/Assets/Scripts/UI/Dynamic/Modals/ConfirmModal.cs
Comment (LucaHaverty) [inline engine/Assets/Scripts/UI/Dynamic/Modals/ConfirmModal.cs:1]: An old panel already exists called StartMatchModePanel that you should delete.

**Review by LucaHaverty — CHANGES_REQUESTED** [review]
Everything looks good, I would just delete the old StartMatchModePanel.cs

**Review by LucaHaverty — APPROVED** [review]
Looks great!

**Review by KyroVibe — CHANGES_REQUESTED** [review]
I'm getting an exception for it trying to switch to a mode that it's already in:
```
New state is the same as the current state (RobotPositioning)
UnityEngine.Debug:LogError (object)
MatchStateMachine:SetState (MatchStateMachine/StateName) (at Assets/Scripts/Modes/MatchStateMachine.cs:38)
MatchStateMachine/MatchConfig/<>c:<Start>b__0_0 () (at Assets/Scripts/Modes/MatchStateMachine.cs:137)
MatchModeModal:<Create>b__7_4 (Synthesis.UI.Dynamic.Button) (at Assets/Scripts/UI/Dynamic/Modals/Spawning/MatchModeModal.cs:43)
Synthesis.UI.Dynamic.Button:<.ctor>b__11_0 () (at Assets/Scripts/UI/Dynamic/DynamicUIComponents.cs:1052)
UnityEngine.EventSystems.EventSystem:Update () (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/EventSystem.cs:530)
```

You can also hit cancel on either the start match or results panel and it kinda just freezes in whatever state that you're in at the time.

**Comment by KyroVibe** [comment]
@PepperLola I'm changing the cancel button bug to a separate issue. You're free to steal it from Luca before he starts it if you want

**Review by Markytt — APPROVED** [review]
Most things work as intended.

**Review by a-crowell — CHANGES_REQUESTED** [review]
Looks good. Cancel buttons work well. Only thing is that somewhere in today's commits the panels got formatted oddly. Happens to all panels in practice mode too.

1553ScoringZonePanel

**Review by a-crowell — APPROVED** [review]
Looks great!

**Review by KyroVibe — CHANGES_REQUESTED** [review]
K I found quite a few issue. I'm not sure if you want to split some of them into separate PRs or just put them all in this one:
1. If you enter match mode from the main menu, it spawns in with no modal popups and no active camera controller.
2. After auto, both robots are controlled at the same time.
3. You can't switch from robot to robot during a match
4. Robots continue to drive when it switches from auto to teleop
5. The camera fixes I added in Luca's branch and are currently in dev for when you begin placing the robot on the field are no longer in your branch.

**Comment by KyroVibe** [comment]
> K I found quite a few issues. I'm not sure if you want to split some of them into separate PRs or just put them all in this one:
> 
>     1. If you enter match mode from the main menu, it spawns in with no modal popups and no active camera controller.
> 
>     2. After auto, both robots are controlled at the same time.
> 
>     3. You can't switch from robot to robot during a match
> 
>     4. Robots continue to drive when it switches from auto to teleop
> 
>     5. The camera fixes I added in Luca's branch and are currently in dev for when you begin placing the robot on the field are no longer in your branch.

So issue 5 is my fault. If you merge dev after my PR, it should be readded.

**Comment by PepperLola** [comment]
4 has always been an issue with disabling behaviours/inputs but not resetting their values and doesn't specifically have to do with this PR, so I think that one should go in a different PR. I'll work on the rest today though.

**Review by KyroVibe — APPROVED** [review]

**Pull request merged**